freqtrade/freqtradebot.py

```diff
@@ -1543,6 +1544,43 @@ def _notify_exit(self, trade: Trade, order_type: str, fill: bool = False) -> Non
         # Send the message
         self.rpc.send_msg(msg)
 
+        open_date = trade.open_date.strftime('%Y-%m-%d %H:%M:%S')
+        close_date = trade.close_date.strftime('%Y-%m-%d %H:%M:%S') if trade.close_date else None
+
+        # Send the message to the discord bot
+        embeds = [{
+            'title': '{} Trade: {}'.format(
+                'Profit' if profit_ratio > 0 else 'Loss',
+                trade.pair),
+            'color': (0x00FF00 if profit_ratio > 0 else 0xFF0000),
+            'fields': [
+                {'name': 'Trade ID', 'value': trade.id, 'inline': True},
+                {'name': 'Exchange', 'value': trade.exchange.capitalize(), 'inline': True},
+                {'name': 'Pair', 'value': trade.pair, 'inline': True},
+                {'name': 'Direction', 'value': 'Short' if trade.is_short else 'Long', 'inline': True},
+                {'name': 'Open rate', 'value': trade.open_rate, 'inline': True},
+                {'name': 'Close rate', 'value': trade.close_rate, 'inline': True},
+                {'name': 'Amount', 'value': trade.amount, 'inline': True},
+                {'name': 'Open order', 'value': trade.open_order_id, 'inline': True},
+                {'name': 'Open date', 'value': open_date, 'inline': True},
+                {'name': 'Close date', 'value': close_date, 'inline': True},
+                {'name': 'Profit', 'value': profit_trade, 'inline': True},
+                {'name': 'Profitability', 'value': '{:.2f}%'.format(profit_ratio * 100), 'inline': True},
+                {'name': 'Stake currency', 'value': self.config['stake_currency'], 'inline': True},
+                {'name': 'Fiat currency', 'value': self.config.get('fiat_display_currency', None), 'inline': True},
+                {'name': 'Buy Tag', 'value': trade.enter_tag, 'inline': True},
+                {'name': 'Sell Reason', 'value': trade.exit_reason, 'inline': True},
+                {'name': 'Strategy', 'value': trade.strategy, 'inline': True},
+                {'name': 'Timeframe', 'value': trade.timeframe, 'inline': True},
+            ],
+        }]
```

Comment:
Letme put this very quickly - not this way.

For notifications - we use the `rpc` package (rpc folder).
If you think notifications for discord are necessary - then this should be implemented as an additional module (similar to telegram or webhooks).

This does not mean that all the same functionalities must be supported (it can be partial integration or whatnot) - but it must follow the same schema (you may also implement only the `send_msg()` method).

This also means, that there will be no changes necessary to other parts of the bot. 
At a first glance, all of these fields should be available in the payloat provided to send_msg as well, therefore i don't see any change necessary to this file.

Reply:
> Is it possible to pass this fields in webhooks?

I assume this should be possible - the webhook defines the complete content that's going to be sent.

From the docs sample:
```
"webhookstatus": {
            "text": "Status: {status}"
        }
```

If you modify this a bit, it could come out as 
```
"webhookstatus": {
            "title": "Trade #{trade_id}, {profit_amount}",
            "fields": ....
        }
```

Now i'm not entirely sure IF/how it works (i never played with this myself) - so if it's not working (or not the way you expect it to), we can either modify things in the webhook to make it work (if it's simple / nonbreaking) - or explicitly support discord (own file / class, subclassing `RPC`) - similar to the current webhook implementation (but designed for discord).

Reply:
I would go with the route of creating separate rpc. For now, will just include these `fill` notification and then we could go on expanding. Wdyt?